File: iosApp/iosApp/Core/UI/Components/AppHeader.swift
import SwiftUI

struct AppHeader: View {
    let text: String

    var body: some View {
        Text(text.uppercased())
            .font(.system(.title2, design: .default, weight: .black))
            .tracking(4)
            .foregroundStyle(Color.accentColor)
            .frame(maxWidth: .infinity, alignment: .center)
    }
}

#Preview("Light") {
    AppHeader(text: "YourApp")
        .padding()
}

#Preview("Dark") {
    AppHeader(text: "YourApp")
        .padding()
        .preferredColorScheme(.dark)
}

File: androidApp/src/main/kotlin/com/yourcompany/kmptemplate/core/ui/components/AppHeader.kt
package com.yourcompany.kmptemplate.core.ui.components

import android.content.res.Configuration.UI_MODE_NIGHT_YES
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.sp
import com.yourcompany.kmptemplate.core.ui.theme.AppTheme
import com.yourcompany.kmptemplate.settings.domain.model.ThemeMode

@Composable
fun AppHeader(text: String, modifier: Modifier = Modifier) {
    Row(
        modifier = modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.Center,
    ) {
        Text(
            text = text.uppercase(),
            style = MaterialTheme.typography.headlineMedium.copy(
                fontWeight = FontWeight.Black,
                letterSpacing = 4.sp,
            ),
            color = MaterialTheme.colorScheme.primary,
        )
    }
}

@Preview(showBackground = true)
@Composable
private fun AppHeaderLightPreview() {
    AppTheme(ThemeMode.LIGHT) { AppHeader(text = "YourApp") }
}

@Preview(showBackground = true, uiMode = UI_MODE_NIGHT_YES)
@Composable
private fun AppHeaderDarkPreview() {
    AppTheme(ThemeMode.DARK) { AppHeader(text = "YourApp") }
}
